src/ert/shared/models/multiple_data_assimilation.py

```diff
@@ -153,18 +173,42 @@ def runSimulations(
         )
         self.setPhaseName(phase_string, indeterminate=True)
 
-        run_context = self.create_context(0, initialize_mask_from_arguments=True)
         update_id = None
         enumerated_weights = list(enumerate(weights))
-        weights_to_run = enumerated_weights[
-            min(self._simulation_arguments["start_iteration"], len(weights)) :
-        ]
-        self.ert().sample_prior(run_context.sim_fs, run_context.active_realizations)
+        starting_iteration = self._simulation_arguments["start_iteration"]
```

Comment:
So this wasn't functioning properly when restarting from an iteration? Because then it would always sample prior?

Reply:
Kind of, though `sample_prior` knows when not to do anything, so it would still work, but it reads horribly and I would like to change that behavior.